stdlib/public/runtime/Heap.cpp

```diff
@@ -28,30 +28,12 @@
 
 using namespace swift;
 
-#if defined(__APPLE__)
-// Apple malloc is always 16-byte aligned.
-#  define MALLOC_ALIGN_MASK 15
-
-#elif defined(__linux__)
-// Linux malloc is 16-byte aligned on 64-bit, and 8-byte aligned on 32-bit.
-#  if defined(__LP64)
-#    define MALLOC_ALIGN_MASK 15
-#  else
-#    define MALLOC_ALIGN_MASK 7
-#  endif
-
-#elif defined(_WIN64)
-// Windows malloc is 16-byte aligned on 64-bit and 8-byte aligned on 32-bit.
-#  define MALLOC_ALIGN_MASK 15
-#elif defined(_WIN32)
-#  define MALLOC_ALIGN_MASK 7
-
-#else
-// Unknown alignment, but the standard requires alignment suitable for the largest
-// standard types.
-#  define MALLOC_ALIGN_MASK std::max(alignof(void *), alignof(double))
-
-#endif
+/// The guaranteed alignment mask for a pointer returned from \c malloc().
+///
+/// The C and C++ standards defined \c max_align_t as a type whose alignment is
+/// at least that of every scalar type. It is the lower bound for the alignment
+/// of any pointer returned from \c malloc().
+static constexpr size_t MALLOC_ALIGN_MASK = alignof(std::max_align_t) - 1;
```

Comment:
Malloc is guaranteed to have at least this alignment, but some systems guarantee more. For example, `alignof(std::max_align_t)` is `8` on 32-bit Darwin, but malloc still gives us 16-byte alignment there. I'd suggest keeping the platform-specific constants and using this construct for the fallback.